packages/vite-runtime-workerd-plugin/src/index.ts

```diff
@@ -4,34 +4,45 @@ import { Miniflare } from 'miniflare';
 
 import { type ViteDevServer } from 'vite';
 import type {
-  SSRRuntime,
   CreateRequestDispatcher,
   DispatchRequest,
+  ServerRuntime,
 } from 'shared-vite-runtime-utils';
 import {
   getFetchModuleUrl,
   setupFetchModuleEndpoint,
+  setupServerRuntimeRegistration,
 } from 'shared-vite-runtime-utils';
 
 export function viteRuntimeWorkerd() {
   return {
     name: 'vite-runtime-workerd-plugin',
-    async configureServer(server: ViteDevServer) {
-      let ssrRuntimeResolve: (runtime: SSRRuntime) => void;
-      const ssrRuntime$ = new Promise<SSRRuntime>(resolve => {
-        ssrRuntimeResolve = resolve;
-      });
-      server.ssrRuntime$ = ssrRuntime$;
+    configureServer(server: ViteDevServer) {
+      // IMPORTANT: the following line should not be needed
+      //            if ViteDevServer were to have the runtime
+      //            registration built in
+      setupServerRuntimeRegistration(server);
+
+      const runtimeCreation = () => {
+        let runtimeResolve: (runtime: ServerRuntime) => void;
+        const runtimePromise = new Promise<ServerRuntime>(resolve => {
+          runtimeResolve = resolve;
+        });
 
-      server.httpServer.once('listening', () => {
-        setupFetchModuleEndpoint(server);
+        server.httpServer.once('listening', () => {
+          setupFetchModuleEndpoint(server);
```

Comment:
observation: since this is a shared helper function, we'll be setting up the endpoint once per server runtime that uses the helper function, but because they override each other, only the first registration will be effective. Ideally the module fetching endpoint is built into Vite proper, but if that's not going to happen, we should modify the implementation of `setupFetchModuleEndpoint` to register the middleware endpoint only once.

Reply:
good point... 🤔 :+1:

Reply:
I've added a simple check in `setupFetchModuleEndpoint` to make sure that we don't register the endpoint more than once 🙂👍